contracts/src/Summa.sol

```diff
@@ -9,163 +9,128 @@ import "@openzeppelin/contracts/access/Ownable.sol";
-     */
-    function setAddressOwnershipVerifier(address _verifier) public onlyOwner {
-        require(_verifier != address(0), "Invalid address verifier");
-        IAddressOwnershipVerifier addressOwnershipVerifier = IAddressOwnershipVerifier(
-                _verifier
-            );
-        bytes32 addressType = addressOwnershipVerifier.getAddressType();
-        require(addressType != bytes32(0), "Invalid address type");
-        cexAddressVerifier[addressType] = addressOwnershipVerifier;
-        emit AddressVerifierSet(addressType, _verifier);
-    }
-
     function submitProofOfAddressOwnership(
-        OwnedAddress[] memory _cexAddresses
+        AddressOwnershipProof[] memory _addressOwnershipProofs
     ) public {
-        for (uint i = 0; i < _cexAddresses.length; i++) {
-            bytes32 addressHash = keccak256(_cexAddresses[i].cexAddress);
-            require(!confirmedAddress[addressHash], "Address already verified");
-            confirmedAddress[addressHash] = true;
+        for (uint i = 0; i < _addressOwnershipProofs.length; i++) {
+            bytes32 addressHash = keccak256(
+                abi.encode(_addressOwnershipProofs[i].cexAddress)
+            );
             require(
-                address(cexAddressVerifier[_cexAddresses[i].addressType]) !=
-                    address(0),
-                "Address verifier not set for this type of address"
+                ownershipProofByAddress[addressHash].signature.length == 0,
+                "Address already verified"
             );
+            ownershipProofByAddress[addressHash] = _addressOwnershipProofs[i];
+            addressOwnershipProofs.push(_addressOwnershipProofs[i]);
             require(
-                cexAddressVerifier[_cexAddresses[i].addressType]
-                    .verifyAddressOwnership(
-                        _cexAddresses[i].cexAddress,
-                        _cexAddresses[i].ownershipProof
-                    ),
-                "Invalid signer"
+                bytes(_addressOwnershipProofs[i].cexAddress).length != 0 &&
+                    bytes(_addressOwnershipProofs[i].chain).length != 0 &&
+                    _addressOwnershipProofs[i].signature.length != 0 &&
+                    _addressOwnershipProofs[i].message.length != 0,
+                "Invalid proof of address ownership"
             );
         }
 
-        emit ExchangeAddressesSubmitted(_cexAddresses);
+        emit ExchangeAddressesSubmitted(_addressOwnershipProofs);
     }
 
     /**
      * @dev Submit proof of solvency for a CEX
-     * @param assets Assets owned by CEX
-     * @param mstRoot The root of the Merkle sum tree
+     * @param mstRoot Merkle sum tree root of the CEX's liabilities
+     * @param assets The list of assets owned by the CEX
      * @param proof The ZK proof
      * @param timestamp The timestamp at which the CEX took the snapshot of its liabilites
      */
     function submitProofOfSolvency(
```

Comment:
I noticed that the submitProofOfSolvency function has been updated to align with an optimistic approach. However, the optimistic approach has not yet been developed for Proof of Ownership.
So, I think it would be helpful to add short notification about the optimistic process that is still a work in progress

Reply:
Done!

Reply:
What do you mean "still work in progress"?

Reply:
I though we have a plan to develop the optimistic approach for Proof of ownership. 
Technically it is not "work in progress".